text/0097-log-data-model.md

```diff
@@ -0,0 +1,968 @@
+conditions both to are TRUE for that record.
+
+### Field: `short_name`
+
+Type: string.
+
+Description: Short event identifier that does not contain varying parts.
+`short_name` describes what happened (e.g. "ProcessStarted"). Recommended to be
+no longer than 50 characters. Optional. Not guaranteed to be unique in any way.
+Typically used for filtering and grouping purposes in backends.
+
+### Field: `body`
+
+Type: any.
+
+Description: A value containing the body of the log record (see the description
+of `any` type above). Can be for example a human-readable string message
+(including multi-line) describing the event in a free form or it can be a
+structured data composed of arrays and maps of other values. Can vary for each
+occurrence of the event coming from the same source.
+
+### Field: `resource`
+
+Type: key/value pair list.
+
+Description: Describes the source of the log, aka
+[resource](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/overview.md#resources).
+"value" of each pair is of `any` type. Multiple occurrences of events coming
+from the same event source can happen across time and they all have the same
+value of `resource`. Can contain for example information about the application
+that emits the record or about the infrastructure where the application runs.
+Data formats that represent this data model may be designed in a manner that
+allows the `resource` field to be recorded only once per batch of log records
+that come from the same source. SHOULD follow OpenTelemetry
+[semantic conventions for Resources](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions).
+
+### Field: `attributes`
+
+Type: key/value pair list.
+
+Description: Additional information about the specific event occurrence. "value"
+of each pair is of `any` type. Unlike the `resource` field, which is fixed for a
+particular source, `attributes` can vary for each occurrence of the event coming
+from the same source. Can contain information about the request context (other
+than TraceId/SpanId). SHOULD follow OpenTelemetry
+[semantic conventions for Attributes](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions).
+
+## Example Log Records
+
+Below are examples that show one possible representation of log records in JSON.
+These are just examples to help understand the data model. Don’t treat the
+examples as _the_ way to represent this data model in JSON.
+
+This document does not define the actual encoding and format of the log record
+representation. Format definitions will be done in separate OTEPs (e.g the log
+records may be represented as msgpack, JSON, Protocol Buffer messages, etc).
+
+Example 1
+
+```json
+{
+  "timestamp": 1586960586000, // JSON needs to make a decision about 
+                              // how to represent nanoseconds.
```

Comment:
Are you expecting the folks who control the JSON standard to make this decision? Or are you suggesting that whoever writes the OTEP for the JSON format should make that decision? Since this example is technically nonconformant, should we toss an option out like `[<milliseconds>, <remaining nanoseconds>]`? Seems like the disclaimer "does not define the actual encoding" should cover any discomfort.

Reply:
Whoever writes an OTEP that defines the representation of this Data Model in JSON format should make that decision.